website/app/core/container.server.ts

```diff
@@ -0,0 +1,125 @@
+      eventKey: process.env.INNGEST_EVENT_KEY,
+    },
+    sessionSecret: process.env.SESSION_SECRET || '',
+    resend: {
+      apiKey: process.env.RESEND_API_KEY,
+    },
+    luma: {
+      apiKey: process.env.LUMA_API_KEY || 'xxx',
+    },
+    zapier: {
+      webhookSecret: process.env.ZAPIER_WEBHOOK_SECRET || 'xxx',
+    },
+    posthog: {
+      publicApiKey: process.env.POSTHOG_PUBLIC_API_KEY || 'xxx',
+    },
+    sentry: {
+      dsn: process.env.SENTRY_DSN,
+      org: process.env.SENTRY_ORG,
+      project: process.env.SENTRY_PROJECT,
+      authToken: process.env.SENTRY_AUTH_TOKEN,
+    },
+  });
+  const logger = createLogger(config.instanceId, ['info', 'debug']);
+
+  // 2. Build the container
+  const container = createContainer<{
+    logger: Logger;
+    queryBus: QueryBus;
+    commandBus: CommandBus;
+    mainConfig: MainConfig;
+    sessionManager: SessionManager;
+    mailer: Mailer;
+    inngestServer: InngestServer;
+    inngestFunctions: ReturnType<typeof buildInngestFunctions>;
+    serverTimingsProfiler: ReturnType<typeof createServerTimingsProfiler>;
+    pocketBaseClient: PocketBaseClient;
+    lumaClient: ReturnType<typeof createLumaClient>;
+    posthogClient: ReturnType<typeof createPosthogClient>;
+    fetchSpeakersWithTalksHandler: ReturnType<typeof createFetchSpeakersWithTalksHandler>;
+    loadEventDetailsHandler: ReturnType<typeof createLoadEventDetailsHandler>;
+  }>({
+    injectionMode: InjectionMode.PROXY,
+    strict: true,
+  });
+
+  container.register({
+    logger: asValue(logger),
+    mainConfig: asValue(config),
+    queryBus: asFunction(() => createQueryBus<QueryDefitions>()).singleton(),
+    commandBus: asFunction(() => createCommandBus<CommandDefitions>()).singleton(),
+    sessionManager: asFunction(createSessionManager)
+      .inject(() => ({ redirect }))
+      .singleton(),
+    mailer: asFunction(createMailer).singleton(),
+    inngestServer: asFunction(createInngestServer).singleton(),
+    inngestFunctions: asFunction(buildInngestFunctions).singleton(),
+    serverTimingsProfiler: asFunction(createServerTimingsProfiler).scoped(),
+    pocketBaseClient: asFunction(createPocketbaseClient).singleton(),
+    lumaClient: asFunction(createLumaClient).singleton(),
+    posthogClient: asFunction(createPosthogClient).singleton(),
+    fetchSpeakersWithTalksHandler: asFunction(createFetchSpeakersWithTalksHandler).singleton(),
+    loadEventDetailsHandler: asFunction(createLoadEventDetailsHandler).singleton(),
+  });
+
+  // 3. Setup the Bus
+  const simpleLogger = { log: container.cradle.logger.info, error: container.cradle.logger.error };
+
+  // Query Bus
+  container.cradle.queryBus.useLoggerMiddleware({ logger: simpleLogger });
+  container.cradle.queryBus.useCacherMiddleware({
+    defaultTtl: 60,
+    defaultStaleTtl: 120,
+    cache: 'all',
+    shortCircuit: true,
+  });
+  container.cradle.queryBus.register('FetchSpeakersWithTalks', container.cradle.fetchSpeakersWithTalksHandler);
```

Comment:
yes we need to setup the Bus

Reply:
Why doesn't every handler have its own bus, wrapping the handler export in its own `domain/` or `modules/` file?

You mentioned one of the motivations for this is that it "scales" once the codebase becomes bigger, but having one bus that needs to implement all requirements of all fetch requests and needs to be the same for all handlers doesn't look like it would scale. It sounds more like Redux or Axios middleware, which historically usually ends up in a big mess with folks adding chaotic code to them.

For instance, how would you solve: 'this handler data should never be cached' or 'this handler, I want to log more than just the default' use cases in the current implementation?

Reply:
The goal of a bus is having a centralized piece of code where you can add middlewares (common to all handlers). 

It is a bit like having multiple express app for every routes ;)

Reply:
of course, here you can have many subfolders in `domain/` to organize with many engineers. same in `contracts/` (here the project is small).

It does scale, there is no doubt about that.

The thing is that middlewares are nice but they are not the solution for everything. Nothing forbids your to repeat some logics in the handler like your would do without a bus. At the end the handler is the simple function that exists right now. Codebase can become a gigantic mess with modules everywhere too.

it's a pattern here, it does not solve everything: https://martinfowler.com/bliki/CQRS.html